server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1038,13 +1039,31 @@ synchronized void completeClose(boolean saveState, boolean completeClose) throws
       activeScan.interrupt();
     }
 
+    // create a copy so that it can be whittled down as client sessions are disabled
+    List<ScanDataSource> runningScans = new ArrayList<>(this.activeScans);
+
+    runningScans.removeIf(scanDataSource -> {
+      boolean disabled = scanDataSource.disableClientSession();
```

Comment:
Yeah that is possible.  I am going to take  a pass at renaming things.  It really initiating disabling the session, going to see if I can name it to reflect that.

Reply:
Updated variable and method names in b6dd442 to better match the behavior of the code.  Also removed a level indirection (a `BooleanSupplier`) I thought I needed to avoid passing around the SessionManager, however found it was not needed.  Removing this level indirection makes the code easier to follow along with the better names.